Fix typos, grammar, and spelling mistakes

BenjaminBrienen opened the pull request:
# Objective

Code and documentation should use words that exist and are spelled correctly.

Also, I noticed a markdownlint warning to do with the table at the bottom having left and right bars (inconsistent style) so I fixed the template file.

## Solution

I used the cSpell VS Code extension and literally opened every single `.rs` file to see if it popped up any `cSpell` "Problems". Most of them were added as "words" in `settings.json`. See here:

```json
"cSpell.words": [
    "0xffffu",
    "Aabb",
    "aabbs",
    "Aaltonen's",
    "aarch",
    "Abney",
    "accesskit",
    "acosf",
    "acosh",
    "acoshf",
    "adjwgt",
    "Adreno",
    "ahash",
    "allocvec",
    "animatable",
    "animgraph",
    "antialiasing",
    "anymap",
    "anytuple",
    "aortiz",
    "APPCOMMAND",
    "arctan",
    "AREATEX",
    "argb",
    "argh",
    "arity",
    "arrayvec",
    "artifacting",
    "asinf",
    "asinh",
    "asinhf",
    "astc",
    "atanf",
    "atanh",
    "atanhf",
    "atomicow",
    "autovectorization",
    "AZERTY",
    "backface",
    "Backfaces",
    "Backquote",
    "backshift",
    "barebones",
    "Barré",
    "barycentrics",
    "batchable",
    "batchables",
    "Bdot",
    "Belen",
    "bevyengine",
    "bevymark",
    "Bezier",
    "Bézier",
    "Beziers",
    "Bezold",
    "Bgra",
    "bilinearly",
    "bincode",
    "bindgen",
    "bindgroup",
    "bindgroups",
    "Bindless",
    "bindpose",
    "bindposes",
    "bitand",
    "bitangent",
    "bitcast",
    "bitfield",
    "Bitfields",
    "bitflags",
    "bitflip",
    "Bitmask",
    "bitsets",
    "bitshifted",
    "bitslice",
    "bitstream",
    "bitvec",
    "BLIT",
    "blobby",
    "bokeh",
    "Borderless",
    "BRDF",
    "Brisebois",
    "Brücke",
    "BSDF",
    "bsparse",
    "Bufferable",
    "bufread",
    "buildable",
    "BURLYWOOD",
    "bytemuck",
    "bytewise",
    "callables",
    "callstack",
    "canonicalize",
    "castable",
    "Castano",
    "Castaño",
    "Casteljau",
    "catmull",
    "cbrt",
    "cbrtf",
    "chacha",
    "cheatbook",
    "chipsets",
    "Choi",
    "CIEXYY",
    "CIEXYZ",
    "cinematics",
    "circumcenter",
    "circumcircle",
    "circumradius",
    "clearcoat",
    "Clippy",
    "Cloneable",
    "Clusterable",
    "cmpeq",
    "cmpge",
    "cmpgt",
    "cmple",
    "CMYKA",
    "codegen",
    "combinators",
    "componentless",
    "componentwise",
    "constness",
    "consts",
    "cooldown",
    "copysign",
    "CORNSILK",
    "cosf",
    "coshf",
    "cpus",
    "criss",
    "crossfade",
    "crossfades",
    "crossfading",
    "cubemap",
    "CUBEMAPS",
    "dasharray",
    "ddsfile",
    "deadzone",
    "dealloc",
    "Deband",
    "debanding",
    "Debouncer",
    "defaultable",
    "denoise",
    "denoised",
    "denoiser",
    "Denoising",
    "denormalization",
    "denormalized",
    "dereferenceable",
    "derefs",
    "desaturate",
    "desaturated",
    "desaturates",
    "desaturation",
    "deserializers",
    "Desmos",
    "despawn",
    "despawned",
    "despawning",
    "despawns",
    "Destructures",
    "DIMAS",
    "discoverability",
    "disocclusion",
    "docsrs",
    "docstring",
    "docstrings",
    "doctest",
    "doubletap",
    "downcasted",
    "downcasting",
    "downsample",
    "downsampling",
    "downscaling",
    "dpad",
    "dropck",
    "Duplo",
    "dvec",
    "Dwarnings",
    "Dxgi",
    "dxil",
    "dylib",
    "easings",
    "Echevarria",
    "Edss",
    "Eevee",
    "ehttpd",
    "Eisu",
    "elerp",
    "ENDCALL",
    "endfor",
    "endianness",
    "ERCIM",
    "evaluateable",
    "expf",
    "expm",
    "eyepatch",
    "fabsf",
    "Farbfeld",
    "Favg",
    "fieldless",
    "finalizer",
    "Fira",
    "fixedbitset",
    "fixint",
    "flac",
    "flatmap",
    "flexbox",
    "fontdb",
    "foos",
    "Footguns",
    "fract",
    "framebuffer",
    "Framelimit",
    "framerates",
    "frametime",
    "freecam",
    "freelist",
    "fullscreen",
    "fullsize",
    "fwidth",
    "FXAA",
    "gabe",
    "GAINSBORO",
    "gamedev",
    "gamepad",
    "gamepads",
    "gameplay",
    "gbuffer",
    "GFBD",
    "Gibibytes",
    "gilrs",
    "Gjøl",
    "Gles",
    "GLSL",
    "Gltf",
    "gltfs",
    "gradlew",
    "graphmap",
    "graymap",
    "grayscale",
    "Greenstein",
    "guillotiere",
    "Habib",
    "Halton",
    "Hanja",
    "Hankaku",
    "hashable",
    "hashbrown",
    "Hasher",
    "hashset",
    "HEADSETHOOK",
    "henkan",
    "Henyey",
    "hexasphere",
    "hittest",
    "HLOD",
    "HLODs",
    "HLSL",
    "hotspot",
    "hoverable",
    "HRTF",
    "Hwba",
    "hypot",
    "hypotf",
    "icosphere",
    "idents",
    "ifdef",
    "ifndef",
    "ilog",
    "IMEs",
    "impls",
    "injective",
    "inlabel",
    "inradius",
    "inscattering",
    "insource",
    "interactable",
    "Internable",
    "INTERNER",
    "interp",
    "irect",
    "irreflexivity",
    "isomp4",
    "itertools",
    "Junja",
    "Kaihuai",
    "karis",
    "Kelemen",
    "keybinds",
    "keycodes",
    "keydown",
    "keysym",
    "keysyms",
    "Khronos",
    "Koschmieder",
    "Kummer",
    "kway",
    "Laba",
    "Lagarde",
    "Lambertian",
    "layouting",
    "Lbottom",
    "Lcha",
    "Ldot",
    "legendaries",
    "Lengyel",
    "lerp",
    "libbevy",
    "libc",
    "libm",
    "libstd",
    "lifetimed",
    "lifetimeless",
    "Lightmap",
    "lightmapped",
    "Lightmapper",
    "lightmaps",
    "LIMEGREEN",
    "linebreak",
    "linebreaks",
    "linestrip",
    "livezone",
    "Lleft",
    "Lleftleft",
    "logcat",
    "logf",
    "łørđ",
    "losslessly",
    "LOTTES",
    "lowerbound",
    "Lright",
    "Lspec",
    "Ltop",
    "Ltoptop",
    "Luma",
    "lumas",
    "luts",
    "Mapface",
    "markdownlint",
    "Masia",
    "memmove",
    "mergeable",
    "Meshable",
    "meshlet",
    "meshlets",
    "meshopt",
    "meshoptimizer",
    "metadatas",
    "metalness",
    "metas",
    "metavar",
    "microfacet",
    "microfacets",
    "midtone",
    "midtones",
    "Mikkelsen's",
    "Mikkelsen",
    "mikktspace",
    "milli",
    "millihertz",
    "minimaps",
    "minimp3",
    "minmax",
    "mipchain",
    "mipmap",
    "mipmapped",
    "Mipmapping",
    "mipmaps",
    "MIRI",
    "mixamo",
    "Moiré",
    "Moissanite",
    "monochromized",
    "monomorphization",
    "monomorphizations",
    "monomorphized",
    "Morten",
    "Morvan",
    "Msaa",
    "msrv",
    "msvc",
    "muhenkan",
    "multimonitor",
    "multiplicatively",
    "Multisample",
    "multisampled",
    "multisampling",
    "multiscatter",
    "Multiscattering",
    "multitouch",
    "multiview",
    "mосква",
    "naïvely",
    "namespacing",
    "nand",
    "Nanite",
    "nanos",
    "natvis",
    "Ndot",
    "NdotV",
    "NEGATIVEX",
    "NEGATIVEY",
    "NEGATIVEZ",
    "Nesw",
    "Neubelt",
    "newtype",
    "NEXTTRACK",
    "nlerp",
    "nocheck",
    "nonmax",
    "nonminimal",
    "Nonnegativity",
    "nonoverlapping",
    "nonplaying",
    "nonprimary",
    "nonsend",
    "nonsendmut",
    "nonspatial",
    "Nout",
    "Numpad",
    "NURB",
    "Nurbs",
    "occluder",
    "OCIO",
    "Oklab",
    "Oklaba",
    "Oklch",
    "Oklcha",
    "OLIVEGREEN",
    "oneshot",
    "openexr",
    "orthogonalization",
    "orthonormalization",
    "orthonormalize",
    "oslog",
    "oversaturates",
    "parallaxed",
    "parallaxing",
    "parameterless",
    "parametrizations",
    "parentless",
    "Pathing",
    "PCSS",
    "PEACHPUFF",
    "peekable",
    "perp",
    "Persson",
    "pessimize",
    "petgraph",
    "Pettineo",
    "Pickable",
    "pickables",
    "PINP",
    "pixmap",
    "Playdead",
    "pointee",
    "pointlight",
    "Polyline",
    "POSITIVEX",
    "POSITIVEY",
    "POSITIVEZ",
    "posterization",
    "postorder",
    "postprocess",
    "powf",
    "powi",
    "powsafe",
    "PQRS",
    "precomputations",
    "precompute",
    "Preedit",
    "Preëdit",
    "Prefilter",
    "preimage",
    "premultiply",
    "prepass",
    "prepasses",
    "preprocess",
    "preprocessable",
    "Preservering",
    "PREVIOUSTRACK",
    "println",
    "PRNG",
    "procmacro",
    "procmacro2_semver_exempt",
    "Proto",
    "proxied",
    "proxying",
    "ptrs",
    "punct",
    "PVRTC",
    "qself",
    "quadro",
    "QWERTZ",
    "Radeon",
    "radsort",
    "RAII",
    "Rampon",
    "Raphson",
    "rasterizer",
    "rasterizing",
    "raycast",
    "Raycasting",
    "RAYMARCH",
    "raymarching",
    "raytrace",
    "raytracing",
    "RCAS",
    "rcvec",
    "readback",
    "readbacks",
    "realloc",
    "REBECCA",
    "rebindings",
    "reborrow",
    "reborrowed",
    "reborrowing",
    "reborrows",
    "recip",
    "recompiles",
    "rects",
    "redepend",
    "Reexit",
    "Reflectable",
    "Reflectstd",
    "reimplementation",
    "reindex",
    "Reinhard",
    "relayout",
    "renderable",
    "renormalize",
    "renormalized",
    "renormalizes",
    "renormalizing",
    "Reparam",
    "reparametrization",
    "reparametrizations",
    "reparametrize",
    "reparametrized",
    "reparent",
    "reparented",
    "reparenting",
    "reparsing",
    "repr",
    "Reproject",
    "reprojected",
    "reprojection",
    "requiree",
    "requirees",
    "reserialize",
    "resmut",
    "respawn",
    "retargeting",
    "retransformed",
    "rgba",
    "Rgba8Snorm",
    "Rgba8Unorm",
    "RGBSDA",
    "rgbw",
    "rngs",
    "rodio",
    "Romaji",
    "roundtripped",
    "roundtrips",
    "Rousiers",
    "RRGGBB",
    "RRGGBBAA",
    "Rrrg",
    "rsplit",
    "runtimes",
    "rustc",
    "rustdoc",
    "RUSTDOCFLAGS",
    "rustfix",
    "RUSTFLAGS",
    "rustfmt",
    "Rustonomicon",
    "rustup",
    "ruzstd",
    "Ryfjallet",
    "samplable",
    "samplr",
    "scancode",
    "scanline",
    "sccs",
    "Schlick",
    "Scratchapixel",
    "screenspace",
    "SEARCHTEX",
    "Seedable",
    "serde",
    "SFLOAT",
    "shadowmap",
    "sharedso",
    "shortnames",
    "Siggraph",
    "signum",
    "simctl",
    "SIMD",
    "simplespace",
    "sincosf",
    "sinf",
    "sinhf",
    "skippable",
    "Şķràźÿ",
    "skybox",
    "skyboxes",
    "slerp",
    "smaa",
    "smallvec",
    "smol_hyper",
    "smol_str",
    "SmolStr",
    "SmolTimer",
    "smoothstep",
    "snorm",
    "Sobotka",
    "spacebar",
    "specializer",
    "specularly",
    "SPIR",
    "SPIRV",
    "splitn",
    "srgb",
    "srgba",
    "SSAA",
    "SSAO",
    "ssbo",
    "ssbos",
    "SSGI",
    "SSRT",
    "Stachowiak",
    "stackfuture",
    "stdcxx",
    "stdlib",
    "steppable",
    "struct",
    "structs",
    "subapp",
    "subasset",
    "subaxis",
    "subcrate",
    "subcrates",
    "subdir",
    "subinterval",
    "subqueries",
    "subquery",
    "Subresource",
    "subsamples",
    "subspan",
    "substate",
    "substates",
    "subtest",
    "SUBTEXTURE",
    "subtrait",
    "subtraits",
    "subtyping",
    "Supercompression",
    "superdump",
    "supersampling",
    "supertrait",
    "supertraits",
    "Svendsen",
    "swapchain",
    "synccell",
    "syncunsafecell",
    "sysinfo",
    "Tanasoaia",
    "tanf",
    "tanhf",
    "Tarjan",
    "Tdot",
    "Teodor",
    "tera",
    "TEXCOORD",
    "texel",
    "texels",
    "textureless",
    "thiserror",
    "thresholded",
    "thresholding",
    "Tiago",
    "Tickrate",
    "tileset",
    "timespan",
    "timestep",
    "timesteps",
    "titlebar",
    "toggleable",
    "Tomasz",
    "TONEMAP",
    "tonemapped",
    "tonemapper",
    "tonemappers",
    "Tonemapping",
    "tony_mc_mapface",
    "TonyMcMapface",
    "topsort",
    "touchpads",
    "transactionality",
    "tspan",
    "turbofish",
    "tweakable",
    "Typeless",
    "UASTC",
    "Ufloat",
    "UI's",
    "uinode",
    "uinodes",
    "uirect",
    "uistack",
    "Ulps",
    "ultrawide",
    "unabbreviate",
    "unaliased",
    "unassigning",
    "unbatchable",
    "unbatchables",
    "unbatched",
    "unboundedness",
    "unclipped",
    "unconstructable",
    "unfixable",
    "unflushed",
    "unforce",
    "ungrab",
    "unhandleable",
    "UNHOVERED",
    "uninit",
    "uniterated",
    "unjittered",
    "unmap",
    "unnormalized",
    "unoccluded",
    "Unorm",
    "unpadded",
    "unproxied",
    "unsampled",
    "Unskewed",
    "unskinned",
    "unspilled",
    "unstretched",
    "unsuffixed",
    "unsynchronized",
    "untaken",
    "untargeted",
    "UNTEXTURED",
    "unthrottled",
    "unwrapper",
    "unyielded",
    "upcasting",
    "upperbound",
    "Upsample",
    "upsampled",
    "upsampling",
    "upscaled",
    "upscaling",
    "urect",
    "ureq",
    "uvec",
    "UYVY",
    "Valorant's",
    "Vals",
    "variadics",
    "Varyings",
    "VBAO",
    "vblank",
    "Vdot",
    "VecDeque",
    "veclike",
    "vecs",
    "verlet",
    "viewports",
    "VIOLETRED",
    "virt",
    "Vlachos",
    "vorbis",
    "voxel",
    "voxelized",
    "voxels",
    "Vsync",
    "VTABLE",
    "Vulkan",
    "VXGI",
    "WAKEUP",
    "WASD",
    "webgl",
    "webgpu",
    "webp",
    "WGPU_SETTINGS_PRIO",
    "wgpu",
    "wgsl",
    "winit",
    "wireframe",
    "wireframes",
    "wlong",
    "wolframalpha",
    "workgroups",
    "worldspace",
    "writeback",
    "wshort",
    "WXYZ",
    "xcodeproj",
    "xcrun",
    "XeGTAO",
    "xffffffff",
    "xmag",
    "xnor",
    "xshell",
    "xvfb",
    "Xwayland",
    "xxyy",
    "xyxy",
    "xyxz",
    "Xyza",
    "xyzw",
    "xyzy",
    "ycbcr",
    "ycocg",
    "yeong",
    "yfov",
    "YIQSDA",
    "YUVSDA",
    "yxwz",
    "Zenkaku",
    "Zeroable",
    "zfar",
    "znear",
    "Zstandard",
    "zstd",
    "παₜα"
  ]
```
## Testing

The CI should be good.

## Migration Guide

`Color::rbga_linear` -> `Color::rgba_linear`
`CubicSegment.coeff` -> `CubicSegment.coefficients`
`RationalSegment.coeff` -> `RationalSegment.coefficients`
`RationalSegment.weight_coeff` -> `RationalSegment.weight_coefficients`
`Curve::reparameterize` -> `Curve::reparametrize`
`Curve::reparametrize_linear` -> `Curve::reparameterize_linear`
`Curve::reparametrize_by_curve` -> `Curve::reparameterize_by_curve`
`FallbackImageMsaa::image_for_samplecount` -> `FallbackImageMsaa::image_for_sample_count`
`prepare_uimaterial_nodes` -> `prepare_ui_material_nodes`
`GridTrack::vmin` -> `GridTrack::v_min`
`GridTrack::vmax` -> `GridTrack::v_max`
`ExitCondition::DontExit` -> `ExitCondition::DoNotExit`
`WinitWindows::get_fitting_videomode` -> `WinitWindows::get_fitting_video_mode`
`get_best_videomode` -> `get_best_video_mode`
`RepeatedGridTrack::vmin` -> `RepeatedGridTrack::v_min`
`RepeatedGridTrack::vmax` -> `RepeatedGridTrack::v_max`
`maths` -> `math`

andresovela commented on crates/bevy_animation/src/animation_curves.rs:
```diff
@@ -1035,7 +1035,7 @@ pub trait AnimationCurveEvaluator: Reflect {
     ///    and the blend register weight to wₘ; then, return success.
     ///
     /// 3. If the blend register is nonempty, call its current value vₙ.
-    ///    Then, set the value of the blend register to vₙ + vₘwₘ.
```
This should be left as it was

rparrett commented on crates/bevy_math/src/cubic_splines.rs:
```diff
@@ -908,30 +908,30 @@ pub trait CyclicCubicGenerator<P: VectorSpace> {
 pub struct CubicSegment<P: VectorSpace> {
     /// Polynomial coefficients for the segment.
-    pub coeff: [P; 4],
+    pub coefficients: [P; 4],
```
Breaking. Good change IMO, but the math people might have opinions.

rparrett commented on crates/bevy_math/src/cubic_splines.rs:
```diff
@@ -1251,9 +1251,9 @@ pub trait RationalGenerator<P: VectorSpace> {
 pub struct RationalSegment<P: VectorSpace> {
     /// The coefficients matrix of the cubic curve.
-    pub coeff: [P; 4],
+    pub coefficients: [P; 4],
```
Breaking

rparrett commented on crates/bevy_math/src/cubic_splines.rs:
```diff
@@ -1251,9 +1251,9 @@ pub trait RationalGenerator<P: VectorSpace> {
+    pub coefficients: [P; 4],
     /// The homogeneous weight coefficients.
-    pub weight_coeff: [f32; 4],
+    pub weight_coefficients: [f32; 4],
```
Breaking

rparrett commented on crates/bevy_render/src/texture/fallback_image.rs:
```diff
@@ -241,7 +241,11 @@ pub struct FallbackImageMsaa<'w> {
 
 impl<'w> FallbackImageMsaa<'w> {
-    pub fn image_for_samplecount(&mut self, sample_count: u32, format: TextureFormat) -> &GpuImage {
+    pub fn image_for_sample_count(
```
Breaking

rparrett commented on crates/bevy_ui/src/render/ui_material_pipeline.rs:
```diff
@@ -424,7 +424,7 @@ pub fn extract_ui_material_nodes<M: UiMaterial>(
 
 #[allow(clippy::too_many_arguments)]
-pub fn prepare_uimaterial_nodes<M: UiMaterial>(
+pub fn prepare_ui_material_nodes<M: UiMaterial>(
```
Breaking

rparrett commented on crates/bevy_ui/src/ui_node.rs:
```diff
@@ -1408,7 +1408,7 @@ impl GridTrack {
 
     /// Create a grid track with a percentage of the viewport's smaller dimension
-    pub fn vmin<T: From<Self>>(value: f32) -> T {
+    pub fn v_min<T: From<Self>>(value: f32) -> T {
```
Breaking, along with other method renames in this file.

rparrett commented on crates/bevy_winit/src/winit_windows.rs:
```diff
@@ -331,7 +331,7 @@ impl WinitWindows {
 ///
 /// The heuristic for "best" prioritizes width, height, and refresh rate in that order.
-pub fn get_fitting_videomode(monitor: &MonitorHandle, width: u32, height: u32) -> VideoModeHandle {
+pub fn get_fitting_video_mode(monitor: &MonitorHandle, width: u32, height: u32) -> VideoModeHandle {
```
Breaking

rparrett commented on crates/bevy_winit/src/winit_windows.rs:
```diff
@@ -362,7 +362,7 @@ pub fn get_fitting_videomode(monitor: &MonitorHandle, width: u32, height: u32) -
 ///
 /// The heuristic for "best" prioritizes width, height, and refresh rate in that order.
-pub fn get_best_videomode(monitor: &MonitorHandle) -> VideoModeHandle {
+pub fn get_best_video_mode(monitor: &MonitorHandle) -> VideoModeHandle {
```
Breaking

rparrett commented on crates/bevy_window/src/lib.rs:
```diff
@@ -189,7 +189,7 @@ pub enum ExitCondition {
     /// create 'headless' processes (processes without windows), which may
     /// surprise your users.
-    DontExit,
+    DoNotExit,
```
Breaking

rparrett commented:
Mostly like these changes, but they should probably be held back until after the 0.15 release cycle is complete.

I am curious as to why `typos` didn't catch some of the more egregious stuff in CI.

BenjaminBrienen commented on crates/bevy_window/src/monitor.rs:
```diff
@@ -31,7 +31,7 @@ pub struct Monitor {
     pub physical_width: u32,
     /// The position of the monitor in physical pixels
     pub physical_position: IVec2,
-    /// The refresh rate of the monitor in millihertz
```
Hertz should be capitalized in plain English, but code doesn't care about proper nouns.

BenjaminBrienen commented on crates/bevy_math/src/curve/mod.rs:
````diff
@@ -427,14 +427,14 @@ pub trait Curve<T> {
     /// ```
     #[must_use]
-    fn reparametrize<F>(self, domain: Interval, f: F) -> ReparamCurve<T, Self, F>
+    fn reparameterize<F>(self, domain: Interval, f: F) -> ReparamCurve<T, Self, F>
````
It is British English spelling, which bevy doesn't use.

BenjaminBrienen commented on docs-template/EXAMPLE_README.md.tpl:
```diff
@@ -303,14 +303,14 @@ may not be worth keeping due to compilation time increases.
-|"z" + lto = "fat"                 | 5.1M     | 9.4M        |
-|"z" + "thin" + codegen-units = 1  | 5.3M     | 11M         |
-|"z" + "fat"  + codegen-units = 1  | 4.8M     | 8.5M        |
+profile                           | wasm-opt | no wasm-opt
```
I'll flip this file around, then.

BenjaminBrienen commented on examples/testbed/3d.rs:
```diff
@@ -138,12 +138,12 @@ mod light {
     }
 }
 
+#[allow(dead_code)]
```
Oh. This generates a warning if you have all features enabled on Windows. I'll remove it.

BenjaminBrienen commented on examples/testbed/3d.rs:
```diff
@@ -191,6 +191,7 @@ mod bloom {
     }
 }
 
+#[allow(dead_code)]
```
Oh. This generates a warning if you have all features enabled on Windows. I'll remove it.

BenjaminBrienen commented:
@rparrett I think I addressed all the feedback. Most of it was adding to the migration guide and applying your suggestions. Let me know if you have further comments! :)

BenjaminBrienen commented:
I have submitted the basic English typos to the typos CI repository.

I can add the long list of exceptions to the `typos.toml` if wanted. I can comment an explanation for each one and I can make sure each one is in the proper category.

atlv24 commented on examples/3d/shadow_biases.rs:
```diff
@@ -135,7 +135,7 @@ fn setup(
                 p.spawn(TextSpan::new("0.0"));
                 p.spawn(TextSpan::new("]\n"));
                 p.spawn(TextSpan::new(
-                    "left/right/up/down/pgup/pgdown - adjust light position (looking at 0,0,0) [",
```
the keys on keyboards literally have it written this way, dont add a space here

atlv24 commented on examples/README.md:
```diff
@@ -92,318 +92,318 @@ git checkout v0.4.0
 
 ## 2D Rendering
 
-Example | Description
```
completely unnecessary changes here revert the entire file

BenjaminBrienen replied:
Our markdown style is to have the tables with that style, so this is just fixing that. It is even inconsistent within the file itself.

BenjaminBrienen replied:
I can make it its own PR if you'd rather.

atlv24 replied:
does it render the same way? i think it looks noisy. can we standardize the other way?

BenjaminBrienen replied:
It does render the same way. I prefer with the bars, and I think that's what is more often used in general. I can change it if a decision-maker wants me to. We only have 2 .md files with tables.

atlv24 commented on examples/ecs/fallible_params.rs:
```diff
@@ -1,5 +1,5 @@
 //! This example demonstrates how fallible parameters can prevent their systems
-//! from running if their acquiry conditions aren't met.
```
acquisition

atlv24 commented on examples/testbed/3d.rs:
```diff
@@ -143,7 +143,6 @@ mod bloom {
         core_pipeline::{bloom::Bloom, tonemapping::Tonemapping},
         prelude::*,
     };
-
```
why

atlv24 commented on examples/tools/scene_viewer/morph_viewer_plugin.rs:
```diff
@@ -43,34 +43,34 @@ const AVAILABLE_KEYS: [MorphKey; 56] = [
-    MorphKey::new("lshift-6", &[KeyCode::ShiftLeft], KeyCode::Digit6),
-    MorphKey::new("lshift-7", &[KeyCode::ShiftLeft], KeyCode::Digit7),
-    MorphKey::new("lshift-8", &[KeyCode::ShiftLeft], KeyCode::Digit8),
-    MorphKey::new("lshift-9", &[KeyCode::ShiftLeft], KeyCode::Digit9),
```
why? this probably breaks things

BenjaminBrienen replied:
"lshift" isn't in the dictionary. Should we add it to the dictionary? I'll run the example to see if it still works.

atlv24 commented on crates/bevy_reflect/derive/src/derive_data.rs:
```diff
@@ -482,7 +482,7 @@ impl<'a> ReflectMeta<'a> {
         )
     }
 
-    /// The collection of docstrings for this type, if any.
```
docstring is a word

BenjaminBrienen replied:
I'll add it to the spelling exceptions, then.

atlv24 commented on crates/bevy_pbr/src/render/pbr_functions.wgsl:
```diff
@@ -125,14 +125,14 @@ fn alpha_discard(material: pbr_types::StandardMaterial, output_color: vec4<f32>)
 // in use (mesh vs. meshlet).
 fn sample_texture(
     texture: texture_2d<f32>,
-    samp: sampler,
```
does this shader even compile anymore

BenjaminBrienen replied:
I changed all instances of the spelling mistake, so I would expect it to be semantically equivalent. I would also expect that compilation errors are caught in CI. If not, maybe we should add that.

atlv24 reviewed (changes requested):
theres a lot of changes i dont like here. beyond some changes being distasteful imo, some changes are just wrong and mangle sentences. the typo fixes are good and very welcome, id hate to see them held back because of being bundled with more controversial changes

the things i dont like:
- lots of commas very liberally added that kinda hurt readability in a lot of cases
- decorating/replacing "so" with "that" and "therefore" and "thus". i dont see the point and it just reads like fluffery
- expansion of understood terms like topsorted to topologically_sorted and coeffs to coefficients, subdivs into subdivisions etc
- changing parametrize into parameterize. both are words but the latter is not the one used in this context in mathematics
- changing NdotV to N_dot_V and similar. this is an existing convention in rendering and this change has been suggested before, we stick to convention here
- im not sure changing pointee to pointed is correct. this isnt my domain but some sentences read a little weird there after the changes
- metallicity -> metalness    why?
- retval -> okay   why?
- birbs -> birds    we're not a corporation we dont have to wear suits and ties
- zfar znear tmin tmax etc underscorification     this hurts readability

i just gave this a skim, i would be very opposed to this merging as is.

id recommend making the objective uncontroversial fixes into a separate PR. id include only the following:
- a -> an changes
- words that are spelled incorrectly (e.g. extra letters, wrong letters, wrong order, wrongly spaced, stuff like strukt -> r#struct)
- conformance to american english (maths -> math, normalise -> normalize)
- dont -> don't and similar apostrophe insertions
- capitalization corrections, including pascal case such as `TestNewcomputedState` => `TestNewComputedState` 
- consistency fixes (having both coeff and coef is weird, lets stick to just one, id pick coeff)
- adding backtick code spans
- unconnected words/sentence fragments like the "You" in crates/bevy_picking/src/lib.rs line 290

i would exclude changes that expand abbreviations, rephrase comments, add commas, or change wordings, and save them for another PR.

maths -> math probably deserves its own PR even

apologies in advance if this review is overly harsh or irritating, i do appreciate the effort to fix this, i just find it a bit overzealous and opinionated

BenjaminBrienen commented on crates/bevy_ui/src/geometry.rs:
```diff
@@ -644,7 +644,7 @@ mod tests {
         let viewport_size = vec2(500., 500.);
 
         for value in (-10..10).map(|value| value as f32) {
-            // for a square viewport there should be no difference between `Vw` and `Vh` and between `Vmin` and `Vmax`.
```
"VMax" isn't in the dictionary but "V" and "Max" are.

atlv24 commented:
also baricenter is not a word, its in the typo exclusions list, not sure where in the repo it is but it should be barycenter

atlv24 commented:
migration guide should include maths -> math too

BenjaminBrienen commented:
> also baricenter is not a word, its in the typo exclusions list, not sure where in the repo it is but it should be barycenter

`crates\bevy_math\src\primitives\dim3.rs`

`#[doc(alias("center", "barycenter", "baricenter"))]`

I can remove the alias

atlv24 commented on examples/3d/shadow_biases.rs:
```diff
@@ -135,7 +135,7 @@ fn setup(
                 p.spawn(TextSpan::new("]\n"));
                 p.spawn(TextSpan::new(
-                    "left/right/up/down/pg up/pg down - adjust light position (looking at 0,0,0) [",
+                    "left/right/up/down/PgUp/PgDown - adjust light position (looking at 0,0,0) [",
```
if you capitalize some capitalize the rest pls

BenjaminBrienen commented:
`metallicity -> metalness` was suggested in another piece of feedback

BenjaminBrienen commented:
Will split into multiple PRs for reviewability based on the lines laid out by atlv. Thanks for the amazing feedback! :)

atlv24 commented on crates/bevy_color/src/color.rs:
```diff
@@ -174,7 +174,7 @@ impl Color {
     #[deprecated = "Use Color::linear_rgba instead."]
     /// Creates a new [`Color`] object storing a [`LinearRgba`] color.
-    pub const fn rbga_linear(red: f32, green: f32, blue: f32, alpha: f32) -> Self {
+    pub const fn rgba_linear(red: f32, green: f32, blue: f32, alpha: f32) -> Self {
```
its deprecated so should be removed anyways. was the original function before deprecation also typo'd?

atlv24 commented on crates/bevy_core_pipeline/src/smaa/smaa.wgsl:
```diff
@@ -64,7 +64,7 @@
  *
  * Note that each [pass] has its own vertex and pixel shader. Remember to use
- * oversized triangles instead of quads to avoid overshading along the
+ * oversized triangles instead of quads to avoid extra shading along the
```
overdraw. this is technically not precisely overdraw but its close enough. also it should say `fullscreen triangle` not `oversized triangles` (not plural)

atlv24 commented on crates/bevy_core_pipeline/src/smaa/smaa.wgsl:
```diff
@@ -444,7 +444,7 @@ const SMAA_LOCAL_CONTRAST_ADAPTATION_FACTOR: f32 = 2.0;
 const SMAA_AREATEX_MAX_DISTANCE_DIAG: f32 = 20.0;
 const SMAA_AREATEX_PIXEL_SIZE: vec2<f32> = (1.0 / vec2<f32>(160.0, 560.0));
-const SMAA_AREATEX_SUBTEX_SIZE: f32 = (1.0 / 7.0);
+const SMAA_AREATEX_SUBTEXTURE_SIZE: f32 = (1.0 / 7.0);
```
this might be texel not texture. id need to read the associated papers or implementations, id leave this alone if i were you. it says TEX in a few other places and id rather stick to whatever convention the papers follow

atlv24 commented on crates/bevy_ecs/src/bundle.rs:
```diff
@@ -374,21 +374,21 @@ impl BundleInfo {
+        deduplicated.sort_unstable();
+        deduplicated.dedup();
 
-        if deduped.len() != component_ids.len() {
+        if deduplicated.len() != component_ids.len() {
             // TODO: Replace with `Vec::partition_dedup` once https://github.com/rust-lang/rust/issues/54279 is stabilized
             let mut seen = HashSet::new();
-            let mut dups = Vec::new();
+            let mut duplicates = Vec::new();
             for id in component_ids {
                 if !seen.insert(id) {
-                    dups.push(id);
+                    duplicates.push(id);
                 }
             }
 
-            let names = dups
+            let names = duplicates
```
id revert this file, dups and deduped is more readable

alice-i-cecile commented:
Closing as split. This PR is too big for me to want to merge.
